Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -478,9 +481,17 @@ void ProgramShaderCache::Shutdown()
 				continue;
 			}
 
-			GLint binary_size;
+			GLint link_status = GL_FALSE, delete_status = GL_TRUE;
+			glGetProgramiv(entry.second.shader.glprogid, GL_LINK_STATUS, &link_status);
+			glGetProgramiv(entry.second.shader.glprogid, GL_DELETE_STATUS, &delete_status);
+			if (glGetError() != GL_NO_ERROR || link_status == GL_FALSE || delete_status == GL_TRUE)
+			{
+				continue;
+			}
+
+			GLint binary_size = 0;
 			glGetProgramiv(entry.second.shader.glprogid, GL_PROGRAM_BINARY_LENGTH, &binary_size);
```

Comment:
I'd merge this two blocks, so call glGetProgramiv three times first, then check for the results.